csrc/ops/composite.cpp

```diff
@@ -299,5 +299,62 @@ TensorView* view_as_real(TensorView* x) {
   auto tv_vector = bitCastOp(vec_type, x);
   return viewAsScalar(tv_vector);
 }
+namespace {
+
+//! Create new output for matmul
+static TensorView* newForMatmul(TensorView* tv_a, TensorView* tv_b) {
+  auto orig_domain_a =
+      TensorDomain::noReductions(tv_a->getMaybeRFactorDomain());
+  auto orig_domain_b =
+      TensorDomain::noReductions(tv_b->getMaybeRFactorDomain());
+
+  auto ndims_a = orig_domain_a.size();
+  auto ndims_b = orig_domain_b.size();
+
+  NVF_ERROR(ndims_a >= 1 && ndims_b >= 1);
+
+  auto ndims_out = std::max(ndims_a, ndims_b);
+  if (std::min(ndims_a, ndims_b) == 1) {
+    ndims_out = std::max(ndims_a, ndims_b) - 1;
+  }
```

Comment:
Can you clarify? It's not immediately clear to me.

Reply:
Added comments. Matmul output size is same as the higher dimensional input size if both A/B > 1D. If one of the inputs is 1D, the output size is 1 less than the higher dimensional input size, since either M/N axis will be missing in the output. For eg, consider: `[M, K] x [K] -> [M]`